frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts

```diff
@@ -195,6 +215,58 @@ export const getEventSourceData = (source: string) => {
   return eventSourceData[source];
 };
 
+export const sanitizeKafkaSourceResource = (formData: EventSourceFormData): EventSourceFormData => {
```

Comment:
It would be good to have test cases for this function to test against invalid values/data types.

Reply:
Have added